pymc/distributions/timeseries.py

```diff
@@ -201,30 +253,39 @@ class GaussianRandomWalk(distribution.Continuous):
         distribution for initial value (Defaults to Flat())
     """
 
-    def __init__(self, tau=None, init=None, sigma=None, mu=0.0, sd=None, *args, **kwargs):
-        kwargs.setdefault("shape", 1)
-        super().__init__(*args, **kwargs)
-        if sum(self.shape) == 0:
-            raise TypeError("GaussianRandomWalk must be supplied a non-zero shape argument!")
-        if sd is not None:
-            sigma = sd
-        tau, sigma = get_tau_sigma(tau=tau, sigma=sigma)
-        self.tau = at.as_tensor_variable(tau)
-        sigma = at.as_tensor_variable(sigma)
-        self.sigma = self.sd = sigma
-        self.mu = at.as_tensor_variable(mu)
-        self.init = init or Flat.dist()
-        self.mean = at.as_tensor_variable(0.0)
-
-    def _mu_and_sigma(self, mu, sigma):
-        """Helper to get mu and sigma if they are high dimensional."""
-        if sigma.ndim > 0:
-            sigma = sigma[1:]
-        if mu.ndim > 0:
-            mu = mu[1:]
-        return mu, sigma
-
-    def logp(self, x):
+    rv_op = gaussianrandomwalk
+
+    @classmethod
+    def dist(
+        cls,
+        mu: Optional[Union[np.ndarray, float]] = None,
+        sigma: Optional[Union[np.ndarray, float]] = None,
+        init: float = None,
+        size: int = None,
+        *args,
+        **kwargs
+    ) -> RandomVariable:
+
+        # Still working on this. The RV op is my current blocker
+        return super().dist([mu, sigma, init], **kwargs)
+
+    # TODO: Add typehints
+    # def get_moment(
+    #     self,
+    #     size: Optional[Union[np.ndarray, float]],
+    #     mu: Optional[Union[np.ndarray, float]],
+    #     sigma: Optional[Union[np.ndarray, float]],
+    #     init: Optional[Union[np.ndarray, float]],
+    # ):
+    #     moment = mu * size + init
+    #     return moment
+
+    def logp(
+        value: Union[at.TensorVariable, at.TensorConstant],
```

Comment:
Did I get this type hint correct now? It was noted that it "almost" always is a symbolic input. Could it be something else?

Reply:
I think you can use the more generic `Variable` which covers both types.

Reply:
> Could it be something else?

It should not

Reply:
> To be really precise here, should it not, or is it not? I ask to ensure the type hint is correct. Updated to at.TensorVariable in meantime

I mean `aesara.graph.basic.Variable` or something like that which works for both `TensorVariables` and `TensorConstants` I think.

The should not thing, is because if users call it directly with numpy arrays it will more often than not work perfectly fine. But internally we always pass symbolic variables and that's what we test in our suite.

Reply:
hm 🤔 
Here's my proposal then. Lets keep it `Union[at.TensorVariable, at.TensorConstant]` Here's my thinking

The things users will see they inspect types `at.Tensorvariable` or `at.TensorConstant`. Even though it would be terser to use the common parent class it would require impicit knowledge of all subclasses that come from that which im not sure that people have

+1 to mypy checks as well. After I get this PR merged I can work on pushing that along.

Reply:
Yeah I agree with you. Seems more readable

Reply:
:tada: